data/api/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/AuthApi.kt

```diff
@@ -9,30 +9,22 @@ import io.ktor.http.ContentType
 import io.ktor.http.HttpHeaders
 import io.ktor.http.HttpStatusCode
 import io.ktor.http.contentType
-import kotlinx.coroutines.flow.first
 
 class AuthApi(
     private val httpClient: HttpClient,
     private val userDataStore: UserDataStore,
     private val authenticator: Authenticator
 ) {
     suspend fun authIfNeeded() {
-        val currentUser = authenticator.currentUser()
-        val isAuthenticated = userDataStore.isAuthenticated().first()
-        val firebaseIdToken = currentUser?.idToken.orEmpty()
-        if (isAuthenticated == true && firebaseIdToken.isNotBlank()) {
-            // already authenticated
-            userDataStore.setIdToken(firebaseIdToken)
-            return
-        }
-        // not authenticated
-        val user = authenticator.signInAnonymously()
-        val createdIdToken = user?.idToken.orEmpty()
-        userDataStore.setIdToken(createdIdToken)
-        if (createdIdToken.isNotBlank()) {
-            registerToServer(createdIdToken)
-            userDataStore.setAuthenticated(true)
+        var currentUser = authenticator.currentUser()
+
+        if (currentUser == null) {
+            // not authenticated
+            currentUser = authenticator.signInAnonymously() ?: return
+            registerToServer(currentUser.idToken)
```

Comment:
Originally, my intention was to try again if Firebase was authenticated but could not be registered with the DroidKaigi API. 👀 
Is it okay to ask for the deleted intention?

Reply:
@takahirom 
I'm sorry, I didn't understand your intent 🙇‍♂️
I have fixed it again. 8c6f8ec

Reply:
I will take a look after I/O extended event 🙇